app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java

```diff
@@ -0,0 +1,155 @@
+    private Button btn_save_filter;
+    private ImageFilter imageFilter;
+
+    private ImageFilterDialogListener listener;
+
+
+    public ImageFilterDialog() {}
+
+
+    public interface ImageFilterDialogListener {
+        public void onFinishDialog(ImageFilter imageFilter);
+    }
+
+    public static ImageFilterDialog newInstance(String title, ImageFilter imageFilter) {
+        ImageFilterDialog dialog = new ImageFilterDialog();
+        Bundle args = new Bundle();
+        args.putString("title",title);
+        args.putSerializable("imageFilter",imageFilter);
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        if (activity instanceof ImageFilterDialogListener) {
+            listener = (ImageFilterDialogListener) activity;
+        } else {
+            throw new ClassCastException(activity.toString()
+                    + " must implement MyListFragment.OnItemSelectedListener");
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_image_filter, container);
+
+        sp_imgsize = (Spinner) view.findViewById(R.id.sp_imgsize);
+        sp_imgcolor = (Spinner) view.findViewById(R.id.sp_imgcolor);
+        sp_img_type = (Spinner) view.findViewById(R.id.sp_img_type);
+        et_img_site = (EditText) view.findViewById(R.id.et_img_site);
+        btn_save_filter = (Button) view.findViewById(R.id.btn_save_filter);
+        sp_imgsize.setFocusable(true);
+
+        if (getArguments().getSerializable("imageFilter") != null) {
+            imageFilter = (ImageFilter) getArguments().getSerializable("imageFilter");
+        }
+        //hideSoftKeyboard(view);
+        setupSpAdapter(view);
+
+        getDialog().getWindow().setSoftInputMode(
+                WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        btn_save_filter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                hideSoftKeyboard(v);
+                finishFilter(v);
+            }
+        });
+
+        return view;
+    }
+
+    public void finishFilter(View view){
+
+        String site = et_img_site.getText().toString();
+        String size = sp_imgsize.getSelectedItem().toString();
+        String color = sp_imgcolor.getSelectedItem().toString();
+        String type = sp_img_type.getSelectedItem().toString();
+
+        imageFilter.setSize(size);
+        imageFilter.setType(type);
+        imageFilter.setColor(color);
+        imageFilter.setSite(site);
+
+        listener.onFinishDialog(imageFilter);
+        dismiss();
+    }
+
+    public void setupSpAdapter(View view){
+
+        ArrayAdapter<CharSequence> adaptercolor = ArrayAdapter.createFromResource(view.getContext(),
```

Comment:
@edwinperaza as your spinners options are static. you are better off defining them within the layout. I'll add a comment on the layout on how you can do that. You do not need to programmatically create this Adapters here.